packages/eslint-plugin-mavenlint/rules/use-flux-standard-actions.js

```diff
@@ -0,0 +1,41 @@
+const standardKeys = ['error', 'meta', 'payload', 'type'];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'enforce using flux standard actions',
+      category: 'Best Practices',
+    },
+    schema: [],
+  },
+  create(context) {
+    return {
+      ReturnStatement(node) {
+        // Is this an action creators file?
+        const path = context.getFilename();
+        if (!path.includes('action-creators')) { return; }
+
+        // Is this return value an Object literal?
+        const statement = node.argument;
+        if (statement.type !== 'ObjectExpression') { return; }
+
+        // Does this Object have a `type` property. If so, infer that this is an action.
+        const properties = statement.properties;
+        const keys = properties.map(property => property.key.name);
+        if (!keys.includes('type')) { return; }
+
+        // Are there non-standard keys?
+        const extraKeys = keys.filter(key => !standardKeys.includes(key));
+        if (extraKeys.length > 0) {
+          context.report({
+            node,
+            message: 'Unexpected keys: {{ unexpectedKeys }}. Generally, redux actions should only have type, payload, error, and meta properties',
+            data: {
+              unexpectedKeys: extraKeys.join(', '),
+            },
+          });
+        }
```

Comment:
Would it be useful to try to optimize lint rules, in general?

Reply:
Definitely seems like that would be useful. We can get timing data from ESLint:

```
Rule                                | Time (ms) | Relative
:-----------------------------------|----------:|--------:
import/no-named-as-default          |  3907.034 |     9.0%
indent                              |  3869.342 |     8.9%
react/void-dom-elements-no-children |  3585.559 |     8.3%
import/no-duplicates                |  3109.513 |     7.2%
import/no-named-as-default-member   |  2046.964 |     4.7%
react/prop-types                    |  1174.247 |     2.7%
react/no-unused-prop-types          |  1158.527 |     2.7%
react/no-deprecated                 |   982.487 |     2.3%
react/prefer-stateless-function     |   982.456 |     2.3%
react/no-string-refs                |   948.852 |     2.2%
```

And this new rule doesn't show up in there. Definitely makes sense to keep track and make linting as fast as possible, @jsteckml

Reply:
Was also able to measure the performance of just the new rule with:

```
TIMING=true yarn lint --no-eslintrc --plugin "mavenlint" --rule "mavenlint/use-flux-standard-actions: 2" --parser "babel-eslint"
```

which generated:

```
Rule                                | Time (ms) | Relative
:-----------------------------------|----------:|--------:
mavenlint/use-flux-standard-actions |    18.027 |   100.0%
```

Not sure if this is telling me the average time spent in the rule or what